keps/sig-docs/20191020-doc-policies-for-third-party-content.md

```diff
@@ -0,0 +1,339 @@
+
+The PR was not approved because SIG Docs didn’t want to add a link to
+third-party content over which SIG Docs have no control.
+
+#### Story 4 (actual)
+
+[Website PR #16203](https://github.com/kubernetes/website/pull/16203)
+removes Stackdriver and Elasticsearch vendor content. Since logging
+falls into the external add-ons category, SIG Docs decided to remove this
+vendor-specific content that had not been meaningfully updated in three
+years.
+
+SIG Docs had buy-in from SIG Instrumentation Bugs for removal; however,
+that PR was held pending the outcome of this KEP and later closed.
+
+#### Story 5 (actual)
+
+In [PR #16766](https://github.com/kubernetes/website/pull/16766)
+@pouledodue proposed adding Hertzner Cloud Controller to the list of
+vendors that have implemented a cloud controller manager. That PR was
+held pending the outcome of this KEP, then later merged.
+
+#### Story 6 (actual)
+
+As [hyperkube transitions to third-party maintenance](https://github.com/kubernetes/kubeadm/issues/1889), it's unclear how to handle [hyperkube content in the Kubernetes docs](https://github.com/kubernetes/website/search?q=hyperkube&unscoped_q=hyperkube) or re-point related links.
+
+### Implementation Details/Notes/Constraints
+
+This KEP originally included language around considering intent of contributors.
+Because intent is effectively impossible to judge (and because contributions
+are nearly always made with the best intent), this KEP now specifies that 
+third-party content is limited to what's required for in-project functionality.
+
+SIG Docs may add its own guidelines for writing and reviewing ambiguous content.
+
+For example:
+> Kubernetes requires out-of-tree software and tools to implement: cluster
+> networking, Ingress, persistent storage, and logging. Hyperlinking to vendor software
+> and documentation _is_ allowed; creating “how to use” content for a specific vendor
+> is discouraged.
+
+Pages that fit with that example guideline:
+ - Cluster Networking
+   - https://kubernetes.io/docs/concepts/cluster-administration/networking/
+   - https://kubernetes.io/docs/concepts/cluster-administration/addons/
+ - Ingress Controllers
+   - https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/#additional-controllers
+ - Persistent Volumes
+   - https://kubernetes.io/docs/concepts/storage/persistent-volumes/#expanding-persistent-volumes-claims
+
+Pages to review and possibly revise, if that guideline were in place:
+ - [Install a Network Policy Provider](https://kubernetes.io/docs/tasks/administer-cluster/network-policy-provider/) and child pages: how to use Calico, Cilium, Kube-router, Romana, and Weave Net for NetworkPolicy
+ - [Audit](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/)
+ - [Use fluentd to collect and distribute audit events from log file](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#use-fluentd-to-collect-and-distribute-audit-events-from-log-file) (dual-sourced)
```

Comment:
This is an interesting one from my perspective. 

It's the only other project on this list officially under the CNCF hood. All other CNCF projects explicitly call out and document their compatibility with Kubernetes. For example, [Vitess](https://vitess.io/) calls out "Run Vitess on Kubernetes" as a top-level menu item in docs. Moreover, there are other CNCF projects like [etcd](https://etcd.io/) which are almost required for running a cluster.

So I think there's two questions we need to think about. One is "What do we do with _third party_ content?" and the second is "Do other CNCF projects count as _third party_?"

Reply:
BTW, Falco is CNCF incubating. It seems to me like it's in a similar position.

Reply:
> "Do other CNCF projects count as third party?"

That seems worth answering. The PR as it stands sees things outside Kubernetes as third party. That's definitely open to revision.

Reply:
@celestehorgan
> "What do we do with third party content?"

Remove it.

> "Do other CNCF projects count as third party?"

They count as third-party (and therefore removable) if they're not necessary for Kubernetes to run in project.

Reply:
@sftim 
> The PR as it stands sees things outside Kubernetes as third party. That's definitely open to revision.

Respectfully, I'd like to limit further revision to this KEP's approvers only. Five months is more than enough time for review. In the case of how we define content, the current language is [two months old](https://github.com/kubernetes/enhancements/pull/1327#issuecomment-579546667), which also seems adequate.